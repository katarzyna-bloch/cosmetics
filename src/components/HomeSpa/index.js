import React, {useState} from 'react'
import PageWrapper from '../PageWrapper'
import ContentHeader from '../ContentHeader'
import ProductList from '../ProductList'

const cosmetics = [
  {
    name: 'Peeling',
    img: 'https://www.topestetic.pl/img/zdjecia/strona-big/resibo-regenerating-body-scrub-lotus-effect-regenerujacy-peeling-do-ciala-300-g.jpg?1613602801',
  },
  {  
    name: 'Szczotka',
    img: 'https://www.topestetic.pl/img/zdjecia/strona-big/mokosh-body-massage-brush-szczotka-do-masazu-ciala-1-szt.jpg?1613602801',
  },
  {
    name: 'Olejek',
    img: 'https://www.topestetic.pl/img/zdjecia/strona-big/veoli-botanica-enjoy-the-calmness-relaksuja-cy-olejek-do-ciala-z-platkami-ro-z-y-150-ml.jpg',
  },
  {
    name: 'Peeling',
    img: 'https://www.topestetic.pl/img/zdjecia/strona-big/resibo-regenerating-body-scrub-lotus-effect-regenerujacy-peeling-do-ciala-300-g.jpg?1613602801',
  },
  {  
    name: 'Szczotka',
    img: 'https://www.topestetic.pl/img/zdjecia/strona-big/mokosh-body-massage-brush-szczotka-do-masazu-ciala-1-szt.jpg?1613602801',
  },
  {
    name: 'Olejek',
    img: 'https://www.topestetic.pl/img/zdjecia/strona-big/veoli-botanica-enjoy-the-calmness-relaksuja-cy-olejek-do-ciala-z-platkami-ro-z-y-150-ml.jpg',
  },
  {
    name: 'Peeling',
    img: 'https://www.topestetic.pl/img/zdjecia/strona-big/resibo-regenerating-body-scrub-lotus-effect-regenerujacy-peeling-do-ciala-300-g.jpg?1613602801',
  },
  {  
    name: 'Szczotka',
    img: 'https://www.topestetic.pl/img/zdjecia/strona-big/mokosh-body-massage-brush-szczotka-do-masazu-ciala-1-szt.jpg?1613602801',
  },
  {
    name: 'Olejek',
    img: 'https://www.topestetic.pl/img/zdjecia/strona-big/veoli-botanica-enjoy-the-calmness-relaksuja-cy-olejek-do-ciala-z-platkami-ro-z-y-150-ml.jpg',
  },
  {
    name: 'Peeling',
    img: 'https://www.topestetic.pl/img/zdjecia/strona-big/resibo-regenerating-body-scrub-lotus-effect-regenerujacy-peeling-do-ciala-300-g.jpg?1613602801',
  },
  {  
    name: 'Szczotka',
    img: 'https://www.topestetic.pl/img/zdjecia/strona-big/mokosh-body-massage-brush-szczotka-do-masazu-ciala-1-szt.jpg?1613602801',
  },
  {
    name: 'Olejek',
    img: 'https://www.topestetic.pl/img/zdjecia/strona-big/veoli-botanica-enjoy-the-calmness-relaksuja-cy-olejek-do-ciala-z-platkami-ro-z-y-150-ml.jpg',
  },
]

function HomeSpa() { 
  const [products, setProducts] = useState(cosmetics)

  return (
    <PageWrapper>
      <ContentHeader>Relaks dla cia≈Ça i duszy</ContentHeader>
      <ProductList products={cosmetics}/>
    </PageWrapper>
  )
}

export default HomeSpa
