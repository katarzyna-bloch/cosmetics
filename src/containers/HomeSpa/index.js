import React, { useEffect } from 'react'
import { connect } from 'react-redux'

import PageWrapper from '../../components/PageWrapper'
import ContentHeader from '../../components/ContentHeader'
import ProductsList from '../../components/ProductsList'
import { fetchHomeSpaProducts } from '../../redux/actions'

export const HomeSpa = ({ fetchProducts, products }) => {
  useEffect(() => {
    fetchProducts()
  }, [])

  return (
    <PageWrapper>
      <ContentHeader>Relaks dla cia≈Ça i duszy</ContentHeader>
      <ProductsList products={products}/>
    </PageWrapper>
  )
}

const mapStateToProps = ({ homeSpaProducts }) => ({
  products: homeSpaProducts,
})

const mapDispatchToProps = dispatch => ({
  fetchProducts: () => dispatch(fetchHomeSpaProducts()),
})

export default connect(mapStateToProps, mapDispatchToProps)(HomeSpa)
