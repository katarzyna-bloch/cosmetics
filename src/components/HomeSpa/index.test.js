import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import HomeSpa from '.'

it('should render correctly', () => {
  let wrapped = shallow(<HomeSpa />)
  expect(toJson(wrapped)).toMatchSnapshot()
})
