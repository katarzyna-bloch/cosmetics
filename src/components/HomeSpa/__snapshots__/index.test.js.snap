// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<styled.div>
  <styled.h5>
    Relaks dla cia≈Ça i duszy
  </styled.h5>
  <ProductList
    products={
      Array [
        Object {
          "img": "https://www.topestetic.pl/img/zdjecia/strona-big/resibo-regenerating-body-scrub-lotus-effect-regenerujacy-peeling-do-ciala-300-g.jpg?1613602801",
          "name": "Peeling",
        },
        Object {
          "img": "https://www.topestetic.pl/img/zdjecia/strona-big/mokosh-body-massage-brush-szczotka-do-masazu-ciala-1-szt.jpg?1613602801",
          "name": "Szczotka",
        },
        Object {
          "img": "https://www.topestetic.pl/img/zdjecia/strona-big/veoli-botanica-enjoy-the-calmness-relaksuja-cy-olejek-do-ciala-z-platkami-ro-z-y-150-ml.jpg",
          "name": "Olejek",
        },
        Object {
          "img": "https://www.topestetic.pl/img/zdjecia/strona-big/resibo-regenerating-body-scrub-lotus-effect-regenerujacy-peeling-do-ciala-300-g.jpg?1613602801",
          "name": "Peeling",
        },
        Object {
          "img": "https://www.topestetic.pl/img/zdjecia/strona-big/mokosh-body-massage-brush-szczotka-do-masazu-ciala-1-szt.jpg?1613602801",
          "name": "Szczotka",
        },
        Object {
          "img": "https://www.topestetic.pl/img/zdjecia/strona-big/veoli-botanica-enjoy-the-calmness-relaksuja-cy-olejek-do-ciala-z-platkami-ro-z-y-150-ml.jpg",
          "name": "Olejek",
        },
        Object {
          "img": "https://www.topestetic.pl/img/zdjecia/strona-big/resibo-regenerating-body-scrub-lotus-effect-regenerujacy-peeling-do-ciala-300-g.jpg?1613602801",
          "name": "Peeling",
        },
        Object {
          "img": "https://www.topestetic.pl/img/zdjecia/strona-big/mokosh-body-massage-brush-szczotka-do-masazu-ciala-1-szt.jpg?1613602801",
          "name": "Szczotka",
        },
        Object {
          "img": "https://www.topestetic.pl/img/zdjecia/strona-big/veoli-botanica-enjoy-the-calmness-relaksuja-cy-olejek-do-ciala-z-platkami-ro-z-y-150-ml.jpg",
          "name": "Olejek",
        },
        Object {
          "img": "https://www.topestetic.pl/img/zdjecia/strona-big/resibo-regenerating-body-scrub-lotus-effect-regenerujacy-peeling-do-ciala-300-g.jpg?1613602801",
          "name": "Peeling",
        },
        Object {
          "img": "https://www.topestetic.pl/img/zdjecia/strona-big/mokosh-body-massage-brush-szczotka-do-masazu-ciala-1-szt.jpg?1613602801",
          "name": "Szczotka",
        },
        Object {
          "img": "https://www.topestetic.pl/img/zdjecia/strona-big/veoli-botanica-enjoy-the-calmness-relaksuja-cy-olejek-do-ciala-z-platkami-ro-z-y-150-ml.jpg",
          "name": "Olejek",
        },
      ]
    }
  />
</styled.div>
`;
